extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java

```diff
@@ -0,0 +1,238 @@
+                if (sourceEntries.stream().anyMatch(entry -> entry.getPath().equals(sourceFullPath))) {
+                    continue;
+                }
+                List<IClasspathAttribute> classpathAttributes = new LinkedList<>();
+                if (isTest) {
+                    classpathAttributes.add(testAttribute);
+                }
+
+                // if the source path is in the build directory, adding optional attribute to it
+                // because it might be cleaned.
+                if (!sourcePath.toFile().exists() || source.getGenerated() || isInBuildDir(sourceFullPath)) {
+                    classpathAttributes.add(optionalAttribute);
+                }
+                sourceEntries.add(JavaCore.newSourceEntry(
+                    sourceFullPath,
+                    null,
+                    null,
+                    outputFullPath,
+                    classpathAttributes.toArray(new IClasspathAttribute[0])
+                ));
+            }
+        }
+        return sourceEntries;
+    }
+
+    private void moveTestTargetsToEnd(List<BuildTarget> buildTargets) {
+        buildTargets.sort((bt1, bt2) -> {
+            return Boolean.compare(bt1.getTags().contains(BuildTargetTag.TEST),
+                bt2.getTags().contains(BuildTargetTag.TEST));
+        });
+    }
+
+    /**
+     * Get the output full path which is used in the classpath entry.
+     */
+    private IPath getOutputFullPath(String outputUri, IProject project) {
+        if (StringUtils.isBlank(outputUri)) {
+            // if output uri is empty, return null to use the default output path.
+            return null;
+        }
+
+        IPath sourceOutputPath = ResourceUtils.filePathFromURI(Utils.getUriWithoutQuery(outputUri).toString());
+        File outputDirectory = sourceOutputPath.toFile();
+        if (!outputDirectory.exists()) {
+            outputDirectory.mkdirs();
+        }
+        IPath relativeSourceOutputPath = sourceOutputPath.makeRelativeTo(project.getLocation());
+        return project.getFolder(relativeSourceOutputPath).getFullPath();
+    }
+
+    /**
+     * Return the first found output uri by kind. Each Gradle build target (source set)
+     * has only one output path for source and resource output.
+     *
+     * @param items the output items.
+     * @param kind the kind of the output uri.
+     */
+    private String getOutputUriByKind(List<OutputPathsItem> items, String kind) {
+        for (OutputPathsItem outputs : items) {
+            for (OutputPathItem output : outputs.getOutputPaths()) {
+                if (Objects.equals(kind, Utils.getQueryValueByKey(output.getUri(), "kind"))) {
+                    return output.getUri();
+                }
+            }
+        }
+        return "";
+    }
+
+    private boolean isInBuildDir(IPath sourceFullPath) {
+        return Arrays.stream(sourceFullPath.segments()).anyMatch(segment -> segment.equals("build"));
```

Comment:
It's a little tricky to check if the folder name contains 'build'.

Reply:
I filed an issue for this case: https://github.com/microsoft/build-server-for-gradle/issues/59

It's doable to get accurate build directory location. I think it's not a very high priority, but I'll do that later.